lib/bundler/resolver.rb

```diff
@@ -344,7 +344,7 @@ def verify_gemfile_dependencies_are_found!(requirements)
           versions = @source_requirements[name][name].map(&:version)
           message  = "Could not find gem '#{requirement}' in #{requirement.source}.\n"
           if versions.any?
-            message << "Source contains '#{name}' at: #{versions.join(", ")}"
+            message << "Source contains '#{name}' at: #{versions.join(", ")} #{requirement.__platform}"
```

Comment:
Hm, okay, have tried the following so far:
```ruby
        build_gem "only_java" do |s|
          s.platform = "java"
          s.write "lib/platform_specific.rb", "PLATFORM_SPECIFIC = '1.0.0 JAVA'"
        end
```
 and

```ruby
        build_gem "only_java" do |s|
          s.platform = "java"
          s.write "lib/only_java.rb", "ONLY_JAVA = '1.0.0 JAVA'"
        end
```

and have not been able to successfully set the platform as java. I might need to dig down the rabbit hole of the test suite gem builder stuff.